sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,556 @@
+ * the credentials information in a struct @ref credman_credential_t. We are
+ * using `CREDMAN_TYPE_PSK` for credman_credential_t::type for PSK credential.
+ *
+ * Next we must assign a @ref credman_tag_t "tag" for the credential. Tags
+ * are unsigned integer value that are used to identify which DTLS sock have
+ * access to which credential. Each DTLS sock will also be assigned a tag.
+ * As a result, a sock can only use credentials that have the same tag as
+ * theirs.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = credman_add(&psk_credential);
+ * if (res < 0) {
+ *     printf("Error cannot add credential to system: %d\n", (int)res);
+ *     return;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After credential infos are filled, we can add it to the credential pool using
+ * `credman_add`.
+ *
+ * Adding credentials of other types are all the same to the above except
+ * credman_credential_t::type and credman_credential_t::param depends on type of
+ * credential used.
+ *
+ * ### Server Operation
+ *
+ * After credentials are added, we can start the server.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include <stdio.h>
+ *
+ * #include "net/sock/dtls.h"
+ *
+ * static uint8_t buf[128];
+ *
+ * int main(void)
+ * {
+ *     // Add credentials
+ *     [...]
+ *
+ *     // initialize server
+ *     sock_udp_t udp_sock;
+ *     sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ *     local.port = 20220;
+ *     sock_udp_create(&udp_sock, &local, NULL, 0);
+ *
+ *     sock_dtls_t dtls_sock;
+ *
+ *     if (sock_dtls_create(&dtls_sock, &udp_sock, DTLS_SOCK_SERVER_TAG, 0) < 0) {
+ *         puts("Error creating DTLS sock");
+ *         return 0;
+ *     }
+ *     sock_dtls_init_server(&dtls_sock);
+ *
+ *     while (1) {
+ *         ssize_t res;
+ *         sock_dtls_session_t session;
+ *
+ *         res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf)),
+ *                              SOCK_NO_TIMEOUT);
+ *         if (res > 0) {
+ *             puts("Received a message");
+ *             if (sock_dtls_send(&dtls_sock, buf, res, &session) < 0) {
+ *                 puts("Error sending reply");
+ *             }
+ *         }
+ *     }
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * This is an example of a DTLS echo server.
+ *
+ * To create a DTLS sock, we first need to have an initialised UDP sock.
+ * DTLS sock will inherit the port from given UDP sock. So the port to be used
+ * later to listen to incoming DTLS packets will need to be set here already.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_t udp_sock;
+ * sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ * local.port = 20220;
+ * sock_udp_create(&udp_sock, &local, NULL, 0);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Using the initialised UDP sock, we can then create our DTLS sock. We use
+ * `DTLS_SOCK_SERVER_TAG`, which is defined as `10` beforehand, as our tag.
```

Comment:
Linking to DTLS_SOCK_SERVER_TAG is not possible here because this macro is locally defined in user application and not in `dtls.h`. Reworded the sentence, mentioned that this value is defined in the user application.